src/Rules/AllowedFile.php

```diff
@@ -118,7 +122,10 @@ public function validate(string $attribute, mixed $value, Closure $fail): void
 
     private function isAllowedExtension(UploadedFile $file): bool
     {
-        $extensions = array_merge(static::EXTENSIONS, config('statamic.assets.additional_uploadable_extensions', []));
+        $extensions = array_merge(
+            $this->allowedExtensions ?? static::EXTENSIONS,
+            config('statamic.assets.additional_uploadable_extensions', [])
+        );
```

Comment:
I would think that if you're passing in allowed extensions, that should be all that's allowed.

In your importer example, you only want to allow csv or xml files. If they add more to the config, then they could upload csv, xml, and whatever else.

Reply:
You're right! 😅

Changed now.